indico/modules/events/registration/controllers/management/invitations.py

```diff
@@ -108,3 +108,23 @@ def _checkParams(self, params):
     def _process(self):
         db.session.delete(self.invitation)
         return jsonify_data(invitation_list=_render_invitation_list(self.regform))
+
+
+class RHRegistrationFormMarkInvitationAsDeclined(RHManageRegFormBase):
```

Comment:
Add a `RHRegistrationFormInvitationBase` that contains checkParams and the normalization spec. Then you can inherit both decline and delete from it.

Instead of checking the current state in the query I'd then simply do this in `_process`:

    if self.invitation.state == InvitationState.pending:
        self.invitation.state = InvitationState.declined